python/cudf/cudf/utils/cudautils.py

```diff
@@ -79,23 +30,27 @@ def find_index_of_val(arr, val, mask=None, compare="eq"):
     mask : mask of the array
     compare: str ('gt', 'lt', or 'eq' (default))
     """
-    found = cuda.device_array(shape=(arr.shape), dtype="int32")
-    if found.size > 0:
-        with _CUDFNumbaConfig():
-            if compare == "gt":
-                gpu_mark_gt.forall(found.size)(arr, val, found, arr.size)
-            elif compare == "lt":
-                gpu_mark_lt.forall(found.size)(arr, val, found, arr.size)
-            else:
-                if arr.dtype in ("float32", "float64"):
-                    gpu_mark_found_float.forall(found.size)(
-                        arr, val, found, arr.size
-                    )
+    arr = cudf.core.column.as_column(arr, nan_as_null=False)
+    target = cudf.core.column.arange(0, arr.size, dtype="int32")
+    if arr.size > 0:
+        if compare == "gt":
+            locations = arr <= val
+        elif compare == "lt":
+            locations = arr >= val
+        else:
+            if arr.dtype in ("float32", "float64"):
+                if np.isnan(val):
+                    locations = is_non_nan(arr)
                 else:
-                    gpu_mark_found_int.forall(found.size)(
-                        arr, val, found, arr.size
-                    )
+                    locations = arr != val
+            else:
+                locations = arr != val
```

Comment:
Could be simplified as?:
```
            if arr.dtype in ("float32", "float64") and np.isnan(val):
                locations = is_non_nan(arr)
            else:
                locations = arr != val
```

Reply:
Do we even need to check the dtype, or is it enough to check `np.isnan(val)`?

Reply:
If so, we could write a one-liner:
```python
locations = is_non_nan(arr) if np.isnan(val) else arr != val
```

Reply:
This works great!